src/polymer/polymer-core-feature-scanner.ts

```diff
@@ -0,0 +1,100 @@
+import * as esutil from '../javascript/esutil';
+import {JavaScriptDocument} from '../javascript/javascript-document';
+import {JavaScriptScanner} from '../javascript/javascript-scanner';
+import {Severity} from '../model/model';
+
+import {toScannedPolymerProperty} from './js-utils';
+import {ScannedPolymerCoreFeature} from './polymer-core-feature';
+
+/**
+ * Scans for Polymer 1.x core "features".
+ *
+ * In the Polymer 1.X core library, the `Polymer.Base` prototype is dynamically
+ * augmented with properties via calls to `Polymer.Base._addFeature`. These
+ * calls are spread across multiple files and split between the micro, mini,
+ * and standard "feature layers". Polymer 2.x does not use this pattern.
+ *
+ * Example: https://github.com/Polymer/polymer/blob/1.x/src/mini/debouncer.html
+ */
+export class PolymerCoreFeatureScanner implements JavaScriptScanner {
+  async scan(
+      document: JavaScriptDocument,
+      visit: (visitor: Visitor) => Promise<void>) {
+    const features: ScannedPolymerCoreFeature[] = [];
+
+    await visit({
+      enterCallExpression(node: estree.CallExpression, parent: estree.Node) {
+        const isAddFeatureCall = esutil.matchesCallExpression(
+            <estree.MemberExpression>node.callee,
+            ['Polymer', 'Base', '_addFeature']);
+        if (!isAddFeatureCall) {
+          return;
+        }
+
+        const description = esutil.getAttachedComment(parent) || '';
+        const feature = new ScannedPolymerCoreFeature(
+            document.sourceRangeForNode(node),
+            node,  // TODO(aomarks) Correct? The astNode param is not typed.
+            description,
+            // TODO(aomarks) Should I set jsdoc?
+            );
+        scanProperties(document, node, feature);
+        features.push(feature);
+      }
+    });
+
+    // TODO(aomarks) Do I need to return warnings here too?
+    return {features: features, warnings: []};
+  }
+}
+
+function scanProperties(
+    document: JavaScriptDocument,
+    node: estree.CallExpression,
+    feature: ScannedPolymerCoreFeature) {
+  if (node.arguments.length !== 1) {
+    feature.warnings.push({
+      message: `Expected only one argument to \`Polymer.Base._addFeature\`. ` +
+          `Got ${node.arguments.length}.`,
+      severity: Severity.ERROR,
+      code: 'invalid-polymer-core-feature-call',
+      sourceRange: document.sourceRangeForNode(node)!,
+    });
+    return;
+  }
+
+  const arg = node.arguments[0];
+  if (arg.type !== 'ObjectExpression') {
+    feature.warnings.push({
+      message: `Expected argument to \`Polymer.Base._addFeature\` to be an ` +
+          `object. Got \`${arg.type}\` instead.`,
+      severity: Severity.ERROR,
+      code: 'invalid-polymer-core-feature-call',
+      sourceRange: document.sourceRangeForNode(node)!,
+    });
+    return;
+  }
+
+  for (const prop of arg.properties) {
+    // TODO(aomarks) Can we have sourceRangeForNode never return undefined?
+    feature.properties.push(
```

Comment:
This needs to be a little more complex to handle methods.
This isn't really the best implementation, but it's code that handles properties, methods, getters and setters: https://github.com/Polymer/polymer-analyzer/blob/master/src/polymer/polymer-element-scanner.ts#L280

It's be nice to clean this up and pull it out so that you _can_ just call one method to handle a property and then see if you get a ScannedProperty or ScannedMethod back.